### DISCUSSION TITLE
Added support for self-hosting custom ports

### DISCUSSION THREAD

**Discussion by xcstatus** [netbirdio/netbird#1406] [Ideas & Feature Requests]
hello all；
    Due to the limitation of network environment, individual users in some places can not obtain port 80, 443. So I wanted to modify the self-hosted deployment script to support custom ports.

TODO：
- [ ] Support for custom ports by manually managing ssl certificates
- [ ] Modify the implementation of caddy ACME challenges to support custom ports, automatic issuance and renewal of ssl certificates

Do you need to implement these contents in the netbird project？

**Comment by xcstatus** [comment]
I have started and implemented the first task.  Some compatibility work is being done to ensure that users using standard ports are not affected by this change.  I'm not sure if I need to support these changes in generic scripts at this point.

**Comment by Darin755** [comment]
I'm verify interested in this as well for self hosting purposes. I have ngnix proxy manager running as well so it would be nice to separate ports.

**Comment by jiangslee** [comment]
https://github.com/netbirdio/netbird/pull/1410

老哥，我把上次的脚本PR上去了，但是我的英文水平有限，沟通起来有点难度。

Brother, I PRed the last script to the official, but my English is relatively poor, so it is a bit difficult to communicate.

**Reply by xcstatus** [reply]
我的英语也是很差。。。
今天我有时间，我在你这一版的基础上做些修改。看看是否可以通过

**Reply by Darin755** [reply]
Can you please communicate in English?

**Reply by jiangslee** [reply]
> Can you please communicate in English?
@Darin755  Sorry, we will use Google Translate to communicate each other.

**Comment by xcstatus** [comment]
https://github.com/netbirdio/netbird/pull/1464

I submitted a PR to fix this issue
